### ISSUE TITLE
Incomplete list of all available plugins

### ISSUE THREAD

**Issue by lukaromih** [opensearch-project/documentation-website#583]
### Is your feature request related to a problem?
OpenSearch has poorly documented all available plugins.

### What solution would you like?
At the minimum, a complete list of all available plugins available to be used with OpenSearch.

### What alternatives have you considered?
The only alternative to poor documentation is frustration. 

### Do you have any additional context?
This issue is the result of this [OpenSearch forum post](https://forum.opensearch.org/t/collation-support/9606).
Short version is: I needed collation support, found a module on Elasticsearch docs, which even described concisely on how to easily install it. This however is NOT Elasticsearch and OpenSearch docs mentioned no such module in its [table of available plugins](https://opensearch.org/docs/latest/opensearch/install/plugins/#available-plugins).

Luckily for me, it turns out OpenSearch HAS support for the module I was after.
When I ran `/usr/share/opensearch/bin/opensearch-plugin list` (lists all installed plugins) inside my OpenSearch Docker container, it returned the list of exactly all the modules listed in the table of available plugins.
When I ran `/usr/share/opensearch/bin/opensearch-plugin install --help` inside the same container, it returned the list of all available official plugins, which also contained the one I was after (analysis-icu). Consequently, I was able to install it pretty easily. So there already are all these other available plugins, ready to be used, and really easily installable, but what good are they, if they're not mentioned anywhere in the docs.
It caused me quite some frustration and dread of not being able to achieve something as vital as language specific alphabetic ordering and the though of having to change the search-engine solution when already so invested into OpenSearch, due to those plugins not being ported from ES to OS, I really don't want other people having to go through it.
The functionality is there, but incomplete documentation results in confusion and I've found other people being confused about it.
I've even found a [PR](https://github.com/opensearch-project/documentation-website/pull/238), where an attempt was made, to better document some of the other modules, but that's for an older version and the results are no longer there and I didn't follow the repo history to find the reasons for their removal.

Determining the best place for the documentation of those missing plugins is left up to the hero who picks up this issue or the community or maintainers.

Also, this issue is just about listing them all. Eventually, I think, their use should also be documented.

**Comment by JeffH-AWS** [comment]
For reference in 2.2:

```
$ opensearch-plugin install -h
Install a plugin

Plugins are packaged as zip files. Each packaged plugin must contain a plugin properties file.

The install command takes a plugin id, which may be any of the following:
  An official opensearch plugin name
  Maven coordinates to a plugin zip
  A URL to a plugin zip
  A local zip file

The following official plugins may be installed by name:
  analysis-icu
  analysis-kuromoji
  analysis-nori
  analysis-phonetic
  analysis-smartcn
  analysis-stempel
  analysis-ukrainian
  discovery-azure-classic
  discovery-ec2
  discovery-gce
  ingest-attachment
  mapper-annotated-text
  mapper-murmur3
  mapper-size
  repository-azure
  repository-gcs
  repository-hdfs
  repository-s3
  store-smb
  transport-nio
```

**Comment by JeffH-AWS** [comment]
@lukaromih thanks for raising this issue.  I'm looking in to it because it also overlaps with another issue that was raised about the docs lacking Maven coordinates for plugins.

Would you like to see a list (the output of `opensearch-plugin install -h`) included for each version?  So for example, for 2.2 it would roughly look like this?

```
opensearch-anomaly-detection         2.2.0.0
opensearch-asynchronous-search       2.2.0.0
opensearch-cross-cluster-replication 2.2.0.0
opensearch-index-management          2.2.0.0
opensearch-job-scheduler             2.2.0.0
opensearch-knn                       2.2.0.0
opensearch-ml                        2.2.0.0
opensearch-observability             2.2.0.0
opensearch-performance-analyzer      2.2.0.0
opensearch-reports-scheduler         2.2.0.0
opensearch-security                  2.2.0.0
opensearch-sql                       2.2.0.0
analysis-icu                         ?.?.?.? // Version numbers would be populated here
analysis-kuromoji                    ?.?.?.? // for the remainder of the plugins
analysis-nori
analysis-phonetic
analysis-smartcn
analysis-stempel
analysis-ukrainian
discovery-azure-classic
discovery-ec2
discovery-gce
ingest-attachment
mapper-annotated-text
mapper-murmur3
mapper-size
repository-azure
repository-gcs
repository-hdfs
repository-s3
store-smb
transport-nio
```

**Comment by JeffH-AWS** [comment]
Findings (work in progress):
```
2.2.0:
analysis-icu
analysis-kuromoji
analysis-nori
analysis-phonetic
analysis-smartcn
analysis-stempel
analysis-ukrainian
discovery-azure-classic
discovery-ec2
discovery-gce
ingest-attachment
mapper-annotated-text
mapper-murmur3
mapper-size
repository-azure
repository-gcs
repository-hdfs
repository-s3
store-smb
transport-nio

2.1.0:
analysis-icu
analysis-kuromoji
analysis-nori
analysis-phonetic
analysis-smartcn
analysis-stempel
analysis-ukrainian
discovery-azure-classic
discovery-ec2
discovery-gce
ingest-attachment
mapper-annotated-text
mapper-murmur3
mapper-size
repository-azure
repository-gcs
repository-hdfs
repository-s3
store-smb
transport-nio

2.0.1:
analysis-icu
analysis-kuromoji
analysis-nori
analysis-phonetic
analysis-smartcn
analysis-stempel
analysis-ukrainian
discovery-azure-classic
discovery-ec2
discovery-gce
ingest-attachment
mapper-annotated-text
mapper-murmur3
mapper-size
repository-azure
repository-gcs
repository-hdfs
repository-s3
store-smb
transport-nio
```
